service/callback.go

```diff
@@ -0,0 +1,59 @@
+package service
+
+import (
+	"fmt"
+	"github.com/gorilla/mux"
+	"net/http"
+
+	"github.com/companieshouse/chs.go/log"
+)
+
+// HandleGovPayCallback handles the callback from Govpay and redirects the user
+func (service *PaymentService) HandleGovPayCallback(w http.ResponseWriter, req *http.Request) {
+	// Get the payment session
+	vars := mux.Vars(req)
+	id := vars["payment_id"]
+	if id == "" {
+		log.ErrorR(req, fmt.Errorf("payment id not supplied"))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// The PaymentResource must be retrieved directly to enable access to metadata outside the data block
+	paymentResource, err := service.DAO.GetPaymentResource(id)
+	if paymentResource == nil {
+		log.ErrorR(req, fmt.Errorf("payment session not found. id: %s", id))
+		w.WriteHeader(http.StatusForbidden)
+		return
+	}
+	if err != nil {
+		log.ErrorR(req, fmt.Errorf("error getting payment resource from db: [%v]", err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Ensure payment method matches endpoint
+	if paymentResource.Data.PaymentMethod == "GovPay" {
```

Comment:
code might be nicer to read if this check is inverted to handle the error case where 
`paymentResource.Data.PaymentMethod != "GovPay"`
then return

means the request processing logic is not nested inside the if statement i.e.

```
        if paymentResource.Data.PaymentMethod != "GovPay" {
		log.ErrorR(req, fmt.Errorf("payment method, [%s], for resource [%s] not recognised", paymentResource.Data.PaymentMethod, id))
		w.WriteHeader(http.StatusBadRequest)
		return
	}
```

then no `else` block is required